pkg/util/http/transport.go

```diff
@@ -55,6 +55,35 @@ func warnOnce(warnMap map[string]bool, key string, format string, params ...inte
 	}
 }
 
+// minTLSVersionFromConfig determines the minimum TLS version defined by the given
+// config, accounting for defaults and deprecated configuration parameters.
+//
+// The returned result is one of the `tls.VersionTLSxxx` constants.
+func minTLSVersionFromConfig(cfg config.Config) uint16 {
+	var min uint16
+	minTLSVersion := cfg.GetString("min_tls_version")
```

Comment:
I think for this one, we may consider evaluating it case-insensitive because users may capitalize `TLS` since that value matches other software capitalization of that constant (e.g. NGINX). If we do make it case-insensitive, I wonder if it also might make sense to use `TLSv1.0` as the default documented value.